superset/migrations/versions/134cea61c5e7_remove_dataset_health_check_message.py

```diff
@@ -0,0 +1,68 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""remove dataset health check message
+
+Revision ID: 134cea61c5e7
+Revises: 301362411006
+Create Date: 2021-04-07 07:21:27.324983
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = "134cea61c5e7"
+down_revision = "301362411006"
+
+import json
+import logging
+
+from alembic import op
+from sqlalchemy import Column, Integer, Text
+from sqlalchemy.ext.declarative import declarative_base
+
+from superset import db
+
+Base = declarative_base()
+
+
+class SqlaTable(Base):
+    __tablename__ = "tables"
+
+    id = Column(Integer, primary_key=True)
+    extra = Column(Text)
+
+
+def upgrade():
+    bind = op.get_bind()
+    session = db.Session(bind=bind)
+
+    for datasource in session.query(SqlaTable):
+        if datasource.extra:
```

Comment:
```suggestion
        if datasource.extra and "health_check" in datasource.extra:
```

This should limit following ops to a smaller subset.

Reply:
This is risky as your checking for the existence of a substring in a JSON encoded string rather than the existence of a key.

Reply:
I don't think it is. It at least saves `json.loads` op for datasources without health_check info, but can be guaranteed to be correct as long as you keep all following steps.

To really save some database IO, we can do the pre-filtering in the SQLA query step:

```python
    for datasource in session.query(SqlaTable).filter(
        SqlaTable.extra.like('%"health_check"%')
    ):
```

Reply:
Granted the downstream logic ensures that the string is actually from a JSON key (I misspoke earlier). That said this migration only takes a few seconds and thus I think the existing logic is fine.

Reply:
Fine by me as well. Just thought it was a good practice to always do pre-filtering when running db migrations.